frontends/web/src/routes/device/bitbox02/checkbackup.tsx

```diff
@@ -14,107 +14,83 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
+import { useState } from 'react';
 import * as bitbox02API from '../../../api/bitbox02';
-import { translate, TranslateProps } from '../../../decorators/translate';
 import { Backup, BackupsListItem } from '../components/backup';
 import { alertUser } from '../../../components/alert/Alert';
 import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
 import { Button } from '../../../components/forms';
+import { useTranslation } from 'react-i18next';
 
-interface CheckProps {
+type TProps = {
     deviceID: string;
     backups: Backup[];
     disabled: boolean;
 }
 
-type Props = CheckProps & TranslateProps;
+const Check = ({ deviceID, backups, disabled }: TProps) => {
+  const [activeDialog, setActiveDialog] = useState(false);
+  const [message, setMessage] = useState('');
+  const [foundBackup, setFoundBackup] = useState<Backup>();
+  const [userVerified, setUserVerified] = useState(false);
+  const { t } = useTranslation();
 
-interface State {
-    activeDialog: boolean;
-    message: string;
-    foundBackup?: Backup;
-    userVerified: boolean;
-}
-
-class Check extends Component<Props, State> {
-  public readonly state: State = {
-    activeDialog: false,
-    message: '',
-    foundBackup: undefined,
-    userVerified: false,
-  };
-
-  private checkBackup = async () => {
-    const { backups, deviceID, t } = this.props;
-    this.setState({ message: t('backup.check.confirmTitle') });
+  const checkBackup = async () => {
```

Comment:
tinynit: you could directly `export const checkBackup = async () => {` here

untested LGTM

Reply:
you meant in the component level `export const Check = ({deviceId, ...})` right?

Reply:
yes I meant export component ofc, thank you   🙈  🙃